src/ufixed.jl

```diff
@@ -31,29 +33,40 @@ const uf14 = UFixedConstructor{UInt16,14}()
 const uf16 = UFixedConstructor{UInt16,16}()
 
 zero{T,f}(::Type{UFixed{T,f}}) = UFixed{T,f}(zero(T),0)
-@generated function one{T<:UFixed}(::Type{T})
-    f = 2^nbitsfrac(T)-1
-    :( T($f,0) )
+function one{T<:UFixed}(::Type{T})
+    T(typemax(rawtype(T)) >> (8*sizeof(T)-nbitsfrac(T)), 0)
```

Comment:
LLVM inlines this new version to a constant:
```jl
julia> @code_llvm one(UFixed{UInt64, 51})

define %UFixed @julia_one_70810(%jl_value_t*) #0 {
top:
  ret %UFixed { i64 2251799813685247 }
}
```
The bigger issue was that for `UInt128` (or even for `UInt64` on 32-bit machines), `2^f` was returning 0 for `f` bigger than the `WORD_SIZE`. This version is guaranteed to be safe no matter what kind of machine or what kind of precision we're using.